### PR TITLE
feat: add "autoUnmount" function

### PR DESCRIPTION
**Pull request by aleclarson**
### Summary

See #234

### Test plan

None yet

### CODE REVIEW

File: src/autoUnmount.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+export default function autoUnmount() {
+  let queue;
+  beforeEach(() => {
+    queue = render.unmountQueue = [];
+  });
+  afterEach(() => {
+    render.unmountQueue = null;
+    queue.forEach(unmount => unmount());
+    queue.length = 0;
+  });
```
Comment (aleclarson) [inline src/autoUnmount.js]: This lets us call `autoUnmount` within a `describe` block to enable auto-unmounting for a subset of tests.

You can also call it from the top-level if you want auto-unmounting for all tests in the file.

**Comment by thymikee** [comment]
Thanks for your PR, appreciate the attitude.

However, I'm not entirely sure about this feature. If you can have this functionality with <10 LOC, why adding it to the library and make us maintain it? I'd rather like to first prove that such a feature is more commonly requested by community first (e.g. in form of many upvotes on the original issue, or a library), then think about implementation. Especially that it's side-effectful and bound to the whole `render` module, making it non composable and available for every test case, even if you don't want it. 

Let's close it for now and keep the discussion going.

**Comment by aleclarson** [comment]
> I'd rather like to first prove that such a feature is more commonly requested by community first

Good point! Another counterpoint is that it's using Jest-specific globals.

> Especially that it's side-effectful and bound to the whole `render` module, making it non composable and available for every test case, even if you don't want it.

Nope, as I tried to clarify in the OP, you can call it inside `describe` blocks too. It unplugs from the `render` module at the end.

**Comment by thymikee** [comment]
> Nope, as I tried to clarify in the OP, you can call it inside describe blocks too. It unplugs from the render module at the end.

Missed that, thanks for point out. Still, I'm not convinced it's a good API and worth getting it into the library, since it can be done entirely in user code. I believe in small API surface area and won't add anything that doesn't add clear value for lots of users.

**Pull request closed without merging**